packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts

```diff
@@ -110,9 +113,18 @@ async function getNetworkId(provider: HttpProvider) {
   const networkIdString = (await provider.request({
     method: "net_version",
   })) as string;
+
   return parseInt(networkIdString, 10);
 }
 
+async function getChainId(provider: HttpProvider) {
+  const chainIdString = (await provider.request({
+    method: "eth_chainId",
```

Comment:
I forgot to mention that this method may not be available. Take a look at how we handle this here: https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/src/internal/core/providers/chainId.ts#L11